### ISSUE TITLE
Passing parameters into the renderer function

### ISSUE THREAD

**Issue by darth-donut** [petrelharp/templater#2]
(From the README)

A correlation with `r n` points:
```{r}
x <- rnorm(n)
y <- x/2+rnorm(n)
plot( x, y )
abline(coef(lm(y~x)))
```
To compile this file, n must be specified beforehand; we could produce ten such output files like so:

```r
for (n in 10*(1:10)) {
    render_template("example_cor.Rmd", output=paste0("cor_",n,".html"))
}
```

Are there plans to mimic `rmarkdown::render`'s `param` parameter?

This is a personal opinion, but explicitly passing parameters (into `rmarkdown::render`'s `param`) seems like a clean way to not "pollute" the `.Rmd` scope.

**Comment by petrelharp** [comment]
> Are there plans to mimic rmarkdown::render's param parameter?
>
> This is a personal opinion, but explicitly passing parameters (into rmarkdown::render's param) seems
> like a clean way to not "pollute" the .Rmd scope.

No, I don't have plans to do this. This is a good idea, though, pull requests welcomed.  I usually do the even cleaner method of running each `render_template` in its own process, via Rscript, like:
```
for N in $(seq 10); do Rscript -e "n <- $N; render_template("example_cor.Rmd", output=paste0("cor_",n,".html")); done
```
